components/modal/src/modal/modal.js

```diff
@@ -28,6 +28,21 @@ export const Modal = ({
     small,
 }) => {
     const layerStyles = resolveLayerStyles(hide)
+
+    const handleKeyDown = (event) => {
+        if (event.key === 'Escape' && onClose) {
+            onClose()
+        }
+    }
+
+    useEffect(() => {
+        document.addEventListener('keydown', handleKeyDown)
```

Comment:
You would probably want to return early when `hide === true`, so that the event listener is not active while its not on the screen.

Reply:
What I mean is something like this:

```js
    useEffect(() => {
        if (hide) {
            return
        }

        const handleKeyDown = (event) => {
            // think it's good practice to check for `defaultPrevented` and respect that
            if (!event.defaultPrevented && event.event.key === 'Escape' && onClose) {
                onClose()
            }
        }
        document.addEventListener('keydown', handleKeyDown)

        return () => {
            document.removeEventListener('keydown', handleKeyDown)
        }
    }, [hide, onClose])
```